code/modules/antagonists/blob/blob.dm

```diff
@@ -47,16 +59,47 @@
 	icon_icon = 'icons/mob/blob.dmi'
 	button_icon_state = "blob"
 
-/datum/action/innate/blobpop/Activate()
+	/// The time taken before this ability is automatically activated.
+	var/autoplace_time = OVERMIND_STARTING_AUTO_PLACE_TIME
+
+/datum/action/innate/blobpop/Grant(Target)
+	. = ..()
+	if(owner)
+		addtimer(CALLBACK(src, .proc/Activate, TRUE), autoplace_time, TIMER_UNIQUE|TIMER_OVERRIDE)
+		to_chat(owner, "<span class='big'><font color=\"#EE4000\">You will automatically pop and place your blob core in [DisplayTimeText(autoplace_time)].</font></span>")
+
+/datum/action/innate/blobpop/Activate(timer_activated = FALSE)
 	var/mob/living/old_body = owner
+	if(!owner)
+		return
+
 	var/datum/antagonist/blob/blobtag = owner.mind.has_antag_datum(/datum/antagonist/blob)
 	if(!blobtag)
-		Remove()
+		Remove(owner)
 		return
+
+	. = TRUE
+	var/turf/target_turf = get_turf(owner)
+	if(target_turf.density)
+		to_chat(owner, "<span class='warning'>This spot is too dense to place a blob core on!</span>")
+		. = FALSE
+	var/area/target_area = get_area(target_turf)
+	if(isspaceturf(target_turf) || !(target_area?.area_flags & BLOBS_ALLOWED) || !is_station_level(target_turf.z))
+		to_chat(owner, "<span class='warning'>You cannot place your core here!</span>")
+		. = FALSE
+
+	var/placement_override = BLOB_FORCE_PLACEMENT
+	if(!.)
+		if(!timer_activated)
+			return
+		placement_override = BLOB_RANDOM_PLACEMENT
+		to_chat(owner, "<span class='boldwarning'>Because your current location is an invalid starting spot and you need to pop, you've been moved to a random location!</span>")
+
 	var/mob/camera/blob/B = new /mob/camera/blob(get_turf(old_body), blobtag.starting_points_human_blob)
```

Comment:
decent name please

Reply:
wasn't really touching that aspect of the code but if you insist.